common/nodejs-utils/src/application.rs

```diff
@@ -55,20 +42,48 @@ pub fn warn_prebuilt_modules(app_dir: &Path, _warn_later: &WarnGuard<Stdout>) {
         log_warning_later(formatdoc! {"
             Warning: {node_modules} checked into source control
 
-            Add these files and directories to {gitignore}. See the Dev Center for more info: 
+            Add these files and directories to {gitignore}. See the Dev Center for more info:
             https://devcenter.heroku.com/articles/node-best-practices#only-git-the-important-bits
         ", node_modules = fmt::value("node_modules"), gitignore = fmt::value(".gitignore") });
     }
 }
 
 #[derive(Debug)]
 pub enum Error {
+    MissingLockfile,
     MultipleLockfiles(Vec<PackageManager>),
 }
 
 impl Display for Error {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
+            Error::MissingLockfile => {
+                writedoc!(
+                    f,
+                    "
+                        Couldn't determine Node.js package manager. Package manager lockfile not found.
+
+                        A lockfile from a supported package manager is required to install Node.js
+                        dependencies. The package.json for this project specifies dependencies, but
+                        there isn't a lockfile.
+
+                        To use npm to install dependencies, run `npm install`. This command will
+                        generate a {npm_lockfile:?} lockfile.
+
+                        Or, to use yarn to install dependencies, run `yarn install`. This command will
+                        generate a {yarn_lockfile:?} lockfile.
+
+                        Or, to use pnpm to install dependencies, run `pnpm install`. This command will
+                        generate a {pnpm_lockfile:?} lockfile.
+
+                        Ensure the resulting lockfile is committed to the repository, then try again.
+                    ",
+                    npm_lockfile = PackageManager::Npm.lockfile(),
```

Comment:
For areas that you want highlighting (e.g.; ``run `npm install`.``) or the filename `{npm_lockfile:?}` then you should  wrap these with the value formatter from the output library:

```rust
npm_install = fmt::value("npm install"),
npm_lockfile = fmt::value(PackageManager::Npm.lockfile().display()),
```

Reply:
Fixed in c823276.